### ISSUE TITLE
DEV: Provide a more descriptive error message for the 'Sending email address' field

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/notification-planning#606]
## Problem statement
[The error message for the 'Sending email address' field on the 'Create a sending email address for your service' didn't describe what the user needed to do to correct their mistake.](https://airtable.com/appWwAmHwDLtpIyko/tbl38n7ofWYBuezFc/viwEUFgBd0KLqH7oo/rechDc3iL4Vw0CJxT?blocks=hide) It's not clear to people that they should not enter an email domain when they complete this field. When people encounter the error, the error message only says "This entry must contain valid characters for an email address" which doesn't describe which characters are valid.

![Screen Shot 2022-04-20 at 10.26.46 AM.png](https://images.zenhubusercontent.com/61267d15419dc0a24aa009dd/fff9ac57-8a18-4283-8b05-14868945843f)

## Hypothesis
1. A more descriptive error message for this field will mean that when a user makes a mistake, they know how to correct their mistake. This can reduce the number of support tickets from people who get stuck while trying to create a service.
2. Better instructions on the page could help the user fill out this form better. We label this field email address, but are only asking for the username part of the address. 

## Acceptance criteria

- [x] Users should know that they only need to create the first part of the email address, before the @ domain, before they make a mistake
- [x] If they make a mistake, the preview shows the corrected characters
- [x] Designs are complete in English and French
- [x] A11y annotations
- [ ] A11y reviewed using auto tools and manual scan on [browser] and [screen reader]
   

## Other considerations

- Most people, especially if they're using assistive tech, will not see the text, "You can change your service email address later in Settings." This may be helpful to include to be provided above the 'Continue' button

## Up to date figma link

https://www.figma.com/file/47FhKMvOXm95IZFL9G8PYN/Creating-bilingual-message-templates-%7C-Creer-des-gabarits-de-message?node-id=2038%3A30161

## Further testing and QA (for implementation card)
- Unit tests for auto correct and validation
- We changed the number of steps
- Both languages look right
- Horizontal overflow of email address

- Link research insight back to hypothesis from Epic or Objective in Airtable
-  Once change/fix/feature is implemented, mark insight as "resolved" in Airtable
-  Once change/fix/feature is implements, link insight to design artifacts (Figma) in Airtable

**Comment by yaelberger-commits** [comment]
@josh-cds for discussion at our next refinement. I think this one is ready to move into the Product Backlog

**Comment by adriannelee** [comment]
I can try to take a look today!

**Comment by adriannelee** [comment]
@andrewleith @amazingphilippe A few questions..

-  Is it intentional that the email domain doesn't get read on the 'Create a sending email address for your service' screen? (The "@staging.notification.cdssandbox.xyz" text)
- Because the error message doesn't tell me what I did wrong, it's hard to know how to correct it. Is that also intentional? In the screenshot below, I'm able to 'pass' while including "gmail.com" The correction works great, but it's not immediately clear what it corrected.
- Now that i'm trying it out, it feels like the "Make sure we formatted your email address correctly" text feels like it shouldn't be in red. It looks like I should change something before clicking Continue, the tone of the message and the colour seem to contradict.
- Should there be an announcement when a user using a screenreader ends up with this 'error'?

![image.png](https://images.zenhubusercontent.com/61267d15419dc0a24aa009dd/c8334c81-d059-4e44-98e7-e5c2ed496ef2)

**Comment by adriannelee** [comment]
This is unlikely .. but when I input "@@@@" and try to submit it, the visible and programatic error messages don't match

![image.png](https://images.zenhubusercontent.com/61267d15419dc0a24aa009dd/f6e53070-eaef-418f-9a31-d77652114c4e)

**Comment by andrewleith** [comment]
> @andrewleith @amazingphilippe A few questions..
> 
> * Is it intentional that the email domain doesn't get read on the 'Create a sending email address for your service' screen? (The "@staging.notification.cdssandbox.xyz" text)

A few questions: 
- Are you using voice over?
- When do you expect it to be read?
- How are you navigating?  Using tab? or VO-Right Arrow?
> * Because the error message doesn't tell me what I did wrong, it's hard to know how to correct it. Is that also intentional? In the screenshot below, I'm able to 'pass' while including "gmail.com" The correction works great, but it's not immediately clear what it corrected.
- I'll let Phil respond to this one.
> * Now that i'm trying it out, it feels like the "Make sure we formatted your email address correctly" text feels like it shouldn't be in red. It looks like I should change something before clicking Continue, the tone of the message and the colour seem to contradict.
- Probably not, but I think we were just piggy-backing on existing functionality.
> * Should there be an announcement when a user using a screenreader ends up with this 'error'?
- The way this error is presented/announced is the same as other fields in the app.  Is a change required across the board?

The idea here was that a user would type something like `My company - mon entreprise` and the app would change it to `my.company-mon.entreprise`, but you bring up some very good cases we should think about.

**Comment by andrewleith** [comment]
> This is unlikely .. but when I input "@@@@" and try to submit it, the visible and programatic error messages don't match
> 

I'm not sure what software you are using there, but this field is tied to two different messages (using `aria-describedby`):
1. Up to 64 characters. Enter what comes before the at symbol, without spaces.
2. Make sure we formatted your email address correctly.

In my testing, Voiceover correctly enumerates both messages when I am focused in the text field.

In this particular case, we "corrected" their text to the point of removing absolutely everything.  I'm not sure if we want to catch this case explicitly or not, but it would be better if it said "This cannot be empty" (i.e. the required validation) or something else a little more specific.

**Comment by amazingphilippe** [comment]
Notes from sync:
Do the following:
- Explain what the domain is, then explain were looking for the name
- Show a complete preview
- Set expectation that you can change it later, before the submit button
- Explain how we formatted "We removed @ and space "

Then
- Decide if we keep the auto fixing
- Does the auto-fixing add enough value for the effort needed in creating an additional page. An additional page is needed to make the interaction good.

**Comment by jimleroyer** [comment]
Phil and Yedida to review this.

**Comment by yaelberger-commits** [comment]
Following design review, bring back into Product backlog for future sprint

**Comment by amazingphilippe** [comment]
Phil has to review and produce new designs

**Comment by amazingphilippe** [comment]
New designs that address the points above: https://www.figma.com/file/47FhKMvOXm95IZFL9G8PYN/Creating-bilingual-message-templates-%7C-Creer-des-gabarits-de-message?node-id=1646%3A28331

We'll have to review the content, but we can start implementing the live preview behaviour. @andrewleith

**Comment by yaelberger-commits** [comment]
Yedida added some questions, sync needed

**Comment by yaelberger-commits** [comment]
Review all screens together for creating a service including this one for consistency,, bilingual information and distinguish where service name appears vs sending email address in From line. Taking to content critique this morning

**Comment by amazingphilippe** [comment]
Card can stay the same. Maybe combine with a new story to keep this organized

**Comment by yaelberger-commits** [comment]
Showed in demo, caught the restriction that you can't use the email in the reply to, annotated. 
Ready for translation!

**Comment by amazingphilippe** [comment]
Translations scheduled for Monday July 4th EOD

**Comment by jimleroyer** [comment]
We got the translations back!

**Comment by amazingphilippe** [comment]
Phil should review and QA this.

**Comment by amazingphilippe** [comment]
Steve and Phil Reviewed for code and UX. Some changes needed Jumana will look at these today
